ems.TestElasticMapsServerCrossNSAssociation fails in cloud-on-k8s-e2e-tests-snapshot-versions-8x job

Job execution https://devops-ci.elastic.co/view/cloud-on-k8s/job/cloud-on-k8s-e2e-tests-snapshot-versions-8x/153/ failed:

```go
=== RUN   TestElasticMapsServerCrossNSAssociation/ElasticMapsServer_Pods_should_eventually_be_ready
Retries (1h0m0s timeout):
    step.go:43: 
        	Error Trace:	utils.go:86
        	Error:      	Received unexpected error:
        	            	pod not running yet
        	Test:       	TestElasticMapsServerCrossNSAssociation/ElasticMapsServer_Pods_should_eventually_be_ready
{"log.level":"error","@timestamp":"2021-05-05T03:44:54.599Z","message":"stopping early","service.version":"0.0.0-SNAPSHOT+00000000","service.type":"eck","ecs.version":"1.4.0","error":"test failure","error.stack_trace":"github.com/elastic/cloud-on-k8s/test/e2e/test.StepList.RunSequential\n\t/go/src/github.com/elastic/cloud-on-k8s/test/e2e/test/step.go:44\ngithub.com/elastic/cloud-on-k8s/test/e2e/ems.TestElasticMapsServerCrossNSAssociation\n\t/go/src/github.com/elastic/cloud-on-k8s/test/e2e/ems/ems_test.go:46\ntesting.tRunner\n\t/usr/local/go/src/testing/testing.go:1193"}
    --- FAIL: TestElasticMapsServerCrossNSAssociation/ElasticMapsServer_Pods_should_eventually_be_ready (3600.00s)
```

Because `docker.elastic.co/elastic-maps-service/elastic-maps-server-ubi8:8.0.0-SNAPSHOT` doesn't exist yet.

```sh
> k -n e2e-venus describe pod test-cross-ns-ems-es-f7rr-ems-7fcfb8cf96-kfwxm | grep Failed
Warning  Failed  37s (x4 over 2m9s)  kubelet  Failed to pull image "docker.elastic.co/elastic-maps-service/elastic-maps-server-ubi8:8.0.0-SNAPSHOT": rpc error: code = Unknown desc = Error response from daemon: manifest for docker.elastic.co/elastic-maps-service/elastic-maps-server-ubi8:8.0.0-SNAPSHOT not found: manifest unknown: manifest unknown
Warning  Failed  37s (x4 over 2m9s)  kubelet  Error: ErrImagePull
Warning  Failed  24s (x6 over 2m9s)  kubelet  Error: ImagePullBackOff
```

--- a/test/e2e/ems/ems_test.go
+++ b/test/e2e/ems/ems_test.go
@@ -13,8 +13,7 @@ import (
 	"github.com/elastic/cloud-on-k8s/test/e2e/test/maps"
 )
 
-// TestElasticMapsServerCrossNSAssociation tests associating Elasticsearch and Elastic Maps Server running in different namespaces.
-func TestElasticMapsServerCrossNSAssociation(t *testing.T) {
+func maybeSkip(t *testing.T) {
 	// only execute this test if we have a test license to work with
 	if test.Ctx().TestLicense == "" {
 		t.SkipNow()
@@ -25,6 +24,15 @@ func TestElasticMapsServerCrossNSAssociation(t *testing.T) {
 	if !stackVersion.GTE(version.MustParse("7.11.0")) {
 		t.SkipNow()
 	}
+	// Elastic Maps Server Docker image 8.0.0-SNAPSHOT does not exist yet
+	if stackVersion.Major >= uint64(8) {
+		t.SkipNow()
+	}
+}
+
+// TestElasticMapsServerCrossNSAssociation tests associating Elasticsearch and Elastic Maps Server running in different namespaces.
+func TestElasticMapsServerCrossNSAssociation(t *testing.T) {
+	maybeSkip(t)
 
 	esNamespace := test.Ctx().ManagedNamespace(0)
 	emsNamespace := test.Ctx().ManagedNamespace(1)
@@ -47,16 +55,7 @@ func TestElasticMapsServerCrossNSAssociation(t *testing.T) {
 }
 
 func TestElasticMapsServerTLSDisabled(t *testing.T) {
-	// only execute this test if we have a test license to work with
-	if test.Ctx().TestLicense == "" {
-		t.SkipNow()
-	}
-
-	stackVersion := version.MustParse(test.Ctx().ElasticStackVersion)
-	// Elastic Maps Server is supported since 7.11.0
-	if !stackVersion.GTE(version.MustParse("7.11.0")) {
-		t.SkipNow()
-	}
+	maybeSkip(t)
 
 	name := "test-ems-tls-disabled"
 
@@ -76,16 +75,7 @@ func TestElasticMapsServerTLSDisabled(t *testing.T) {
 }
 
 func TestElasticMapsServerVersionUpgradeToLatest7x(t *testing.T) {
-	// only execute this test if we have a test license to work with
-	if test.Ctx().TestLicense == "" {
-		t.SkipNow()
-	}
-
-	stackVersion := version.MustParse(test.Ctx().ElasticStackVersion)
-	// Elastic Maps Server is supported since 7.11.0
-	if !stackVersion.GTE(version.MustParse("7.11.0")) {
-		t.SkipNow()
-	}
+	maybeSkip(t)
 
 	srcVersion := test.Ctx().ElasticStackVersion
 	dstVersion := test.LatestVersion7x

--- a/test/e2e/ems/ems_test.go
+++ b/test/e2e/ems/ems_test.go
@@ -7,33 +7,13 @@ package ems
 import (
 	"testing"
 
-	"github.com/elastic/cloud-on-k8s/pkg/controller/common/version"
 	"github.com/elastic/cloud-on-k8s/test/e2e/test"
 	"github.com/elastic/cloud-on-k8s/test/e2e/test/elasticsearch"
 	"github.com/elastic/cloud-on-k8s/test/e2e/test/maps"
 )
 
-func maybeSkip(t *testing.T) {
-	// only execute this test if we have a test license to work with
-	if test.Ctx().TestLicense == "" {
-		t.SkipNow()
-	}
-
-	stackVersion := version.MustParse(test.Ctx().ElasticStackVersion)
-	// Elastic Maps Server is supported since 7.11.0
-	if !stackVersion.GTE(version.MustParse("7.11.0")) {
-		t.SkipNow()
-	}
-	// Elastic Maps Server Docker image 8.0.0-SNAPSHOT does not exist yet
-	if stackVersion.Major >= uint64(8) {
-		t.SkipNow()
-	}
-}
-
 // TestElasticMapsServerCrossNSAssociation tests associating Elasticsearch and Elastic Maps Server running in different namespaces.
 func TestElasticMapsServerCrossNSAssociation(t *testing.T) {
-	maybeSkip(t)
-
 	esNamespace := test.Ctx().ManagedNamespace(0)
 	emsNamespace := test.Ctx().ManagedNamespace(1)
 	name := "test-cross-ns-ems-es"
@@ -55,8 +35,6 @@ func TestElasticMapsServerCrossNSAssociation(t *testing.T) {
 }
 
 func TestElasticMapsServerTLSDisabled(t *testing.T) {
-	maybeSkip(t)
-
 	name := "test-ems-tls-disabled"
 
 	esBuilder := elasticsearch.NewBuilder(name).
@@ -75,8 +53,6 @@ func TestElasticMapsServerTLSDisabled(t *testing.T) {
 }
 
 func TestElasticMapsServerVersionUpgradeToLatest7x(t *testing.T) {
-	maybeSkip(t)
-
 	srcVersion := test.Ctx().ElasticStackVersion
 	dstVersion := test.LatestVersion7x
 

--- a/test/e2e/test/maps/builder.go
+++ b/test/e2e/test/maps/builder.go
@@ -148,6 +148,17 @@ func (b Builder) MutationReversalTestContext() test.ReversalTestContext {
 }
 
 func (b Builder) SkipTest() bool {
+	// only execute EMS tests if we have a test license to work with
+	if test.Ctx().TestLicense == "" {
+		return true
+	}
+
+	// TODO: remove once an EMS 8.x Docker image image is available
+	stackVersion := version.MustParse(test.Ctx().ElasticStackVersion)
+	if stackVersion.Major >= uint64(8) {
+		return true
+	}
+
 	ver := version.MustParse(b.EMS.Spec.Version)
 	return version.SupportedMapsVersions.WithinRange(ver) != nil
 }

--- a/test/e2e/test/maps/builder.go
+++ b/test/e2e/test/maps/builder.go
@@ -154,6 +154,7 @@ func (b Builder) SkipTest() bool {
 	}
 
 	// TODO: remove once an EMS 8.x Docker image image is available
+	// https://github.com/elastic/cloud-on-k8s/issues/4479
 	stackVersion := version.MustParse(test.Ctx().ElasticStackVersion)
 	if stackVersion.Major >= uint64(8) {
 		return true

